source/input_types.rst

```diff
@@ -0,0 +1,192 @@
+
+.. _integer:
+
+integer
+-------
+A whole number (positive or negative). Integers are often used to uniquely identify something.
+Because rasters can only store numeric data, it is common to use integer codes as labels that map to non-numeric data
+(such as land use/landcover types and soil groups).
+
+.. _text:
+
+text
+----
+Freeform text. InVEST accepts any Unicode character. For best results, use Unicode character sets for non-Latin alphabets.
+
+.. _option:
+
+option
+------
+A choice from a fixed set of options. Options are usually displayed as a dropdown menu. Some CSV inputs have columns that must contain values from a fixed set of options. In that case, be sure to enter the options exactly as they are written.
+
+
+.. _truefalse:
+
+true/false
+----------
+True/false inputs are usually displayed as a checkbox. Some CSV inputs have true/false columns. In that case, the model's documentation will explain exactly how to enter the values (``TRUE`` or ``FALSE``, ``1`` or ``0``, etc.)
+
+
+.. _csv:
+
+CSV
+---
+A table of data stored in a .csv (comma-separated value) format. You may use any spreadsheet software to create and edit a table, and then export it to CSV format.
+
+
+.. _raster:
+
+raster
+------
+A spatial data file that stores data in pixels, which each have a numeric value.
+InVEST accepts all GDAL-supported raster formats, including GeoTiff (.tif) and many more.
+For the complete list, see https://gdal.org/drivers/raster/index.html.
+
+Bands
+~~~~~
+Rasters may have multiple bands. All InVEST models look at the first band only. If you are using a multi-band raster,
+please be sure that the correct dataset is in the first band.
+
+
+.. _datatypes:
+
+Data types
+~~~~~~~~~~
+Every raster has a *data type* which determines the minimum and maximum value that each pixel can have. Some data types allow positive and negative numbers, while others only allow positive numbers. Most of the time you will not need to change your raster's data type, but it is important to be aware of.
+
+Understanding data types
+^^^^^^^^^^^^^^^^^^^^^^^^
+A raster's data type is displayed as a code in the raster metadata, which may be viewed in your GIS software. A data type code has 3 components:
+
+1. Unsigned marker (**u**), optional
+
+   By default, data types are signed, meaning that they include a sign (+ or -) indicating whether the number is positive or negative.
+   You need a signed data type to store negative data.
+   If a data type begins with a **u**, that means it is unsigned. All unsigned data is positive. If you do not need to store negative data,myou can save space by using an unsigned type.
+
+   This distinction only exists for integer data types. Float types are always signed.
+
+2. Type (**float** or **int**)
+
+   Floating-point (float) types can store digits after the decimal point. There is no hard limit on how many decimal places they can store, but they are only accurate to a limited number of decimal places.
```

Comment:
I'm not completely sure we need to get fully into this in this document, but there is also a [limit to the accuracy of the _integers_](https://en.wikipedia.org/wiki/Single-precision_floating-point_format#Precision_limitations_on_integer_values) the floating-point spec can represent, not just decimal precision.  That's the tradeoff with IEEE754 ... it can approximate a massive range of numbers, but we lose precision the farther away from 0 we get.

I think the important tidbit about integers is that if you need integer precision > 2^24, folks should use uint32.

Reply:
Oh that's wacky!
```
>>> numpy.array([16777217], dtype=numpy.int32)
array([16777217], dtype=int32)
>>> numpy.array([16777217], dtype=numpy.float32)
array([16777216.], dtype=float32)
```
So maybe the precision of 6 or 7 digits is really _total_ digits, before or after the decimal point?
```
2^0+2^−23 =         1.0000001...
2^4+2^−19 =        16.000001...
2^7+2^-16 =       128.00001...
2^10+2^-13 =     1024.0001...
2^14+2^-9 =     16384.001...
2^17+2^-6 =    131072.01...
2^20+2^-3 =   1048576.1...
2^24+2^1 =   1677721x.
2^27+2^4 =  1342177xx.
2^30+2^7 = 1073741xxx.

Reply:
I added a note to clarify this a little. I don't think we need to go into much detail.

Reply:
Yes, the design of single-precision floating-point numbers is to have about 7 significant decimal digits.  Double-precision floating-point numbers have about 16 significant decimal digits.